libvast/src/system/sink.cpp

```diff
@@ -74,13 +74,18 @@ transforming_sink(caf::stateful_actor<sink_state>* self,
         VAST_INFO("{} received first result with a latency of {}",
                   self->state.name, to_string(time_since_flush));
       }
-      auto transformed = self->state.transforms.apply(std::move(slice));
+      auto add_failed = self->state.transforms.add(std::move(slice));
+      if (add_failed) {
+        VAST_ERROR("add failed {}", add_failed);
+        return;
+      }
+      auto transformed = self->state.transforms.finish();
       if (!transformed) {
         VAST_WARN("discarding slice; error in output transformation: {}",
                   transformed.error());
         return;
       }
-      slice = std::move(*transformed);
+      slice = std::move((*transformed)[0]);
```

Comment:
This smells like a bug waiting to be introduced to me. This currently works because you know you only get a single slice back, but why not just do a two-step change here:
1. Check if the result was empty
2. Iterate over all slices in the result

Reply:
Yeah, this was for an intermediate rewrite step. We caught all other instances, but this one seems to have been missing a FIXME.